command/agent/config_parse.go

```diff
@@ -327,10 +328,14 @@ func extraKeys(c *Config) error {
 		helper.RemoveEqualFold(&c.ExtraKeysHCL, "telemetry")
 	}
 
-	// The `vault` and `consul` blocks are parsed separately from the Decode method, so it
+	// Remove reporting extra keys
+	c.ExtraKeysHCL = slices.DeleteFunc(c.ExtraKeysHCL, func(s string) bool { return s == "license" })
+
+	// The`vault` and `consul` blocks are parsed separately from the Decode method, so it
 	// will incorrectly report them as extra keys, of which there may be multiple
 	c.ExtraKeysHCL = slices.DeleteFunc(c.ExtraKeysHCL, func(s string) bool { return s == "vault" })
 	c.ExtraKeysHCL = slices.DeleteFunc(c.ExtraKeysHCL, func(s string) bool { return s == "consul" })
```

Comment:
```suggestion
	// The following blocks are parsed separately from the Decode method, so they must be removed from extra
	// keys to avoid incorrect report.
	c.ExtraKeysHCL = slices.DeleteFunc(c.ExtraKeysHCL, func(s string) bool { return s == "license" })
	c.ExtraKeysHCL = slices.DeleteFunc(c.ExtraKeysHCL, func(s string) bool { return s == "vault" })
	c.ExtraKeysHCL = slices.DeleteFunc(c.ExtraKeysHCL, func(s string) bool { return s == "consul" })
```

I think these are all removed for the same reason.

Reply:
I thought the same, but Tim corrected me, for vault and consul they are different because multiple blocks are allowed, so they are their own special case